packages/webapp/src/components/EnvironmentPicker.tsx

```diff
@@ -0,0 +1,92 @@
+import { Popover, PopoverContent, PopoverTrigger } from '../components/ui/Popover';
+import { Command, CommandEmpty, CommandGroup, CommandInput, CommandItem, CommandList } from '../components/ui/Command';
+import { useMeta } from '../hooks/useMeta';
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import { IconCheck, IconChevronDown } from '@tabler/icons-react';
+import { Button } from './ui/button/Button';
+import { useStore } from '../store';
+import { cn } from '../utils/utils';
+import { useEnvironment } from '../hooks/useEnvironment';
+
+export const EnvironmentPicker: React.FC = () => {
+    const navigate = useNavigate();
+
+    const env = useStore((state) => state.env);
+    const setEnv = useStore((state) => state.setEnv);
+
+    const { meta } = useMeta();
+    const { mutate } = useEnvironment(env);
+    const [open, setOpen] = useState(false);
+
+    const onSelect = (selected: string) => {
+        if (selected === env) {
+            return;
+        }
+
+        setEnv(selected);
+        void mutate();
+
+        const pathSegments = window.location.pathname.split('/').filter(Boolean);
+
+        pathSegments[0] = selected;
+
+        let newPath = `/${pathSegments.join('/')}`;
+
+        // If on 'integration' or 'connections' subpages beyond the second level, redirect to their parent page
+        if (pathSegments[1] === 'integrations' && pathSegments.length > 2) {
+            newPath = `/${selected}/integrations`;
+        } else if (pathSegments[1] === 'connections' && pathSegments.length > 2) {
+            newPath = `/${selected}/connections`;
+        }
+
+        navigate(newPath);
```

Comment:
Should the popover close when selecting an item?

Reply:
I found that handy to quickly view stuff from different environments, and when you click the wrong one you don't need to reopen it. Don't have a strong opinion.
@bastienbeurier what do you think

https://github.com/user-attachments/assets/7bc5d599-fc37-4a9f-b0e2-9a041014eb95

Reply:
I have no strong feelings about this; I think both can work. I would say priority to the implementer's opinion :D